### DISCUSSION TITLE
About the source

### DISCUSSION THREAD

**Discussion by JAJA-WHB** [audacity/audacity#2994] [Development]
How is your noise reduction function realized?

**Comment by rbdannenberg** [comment]
If you mean the algorithm, it's basically a noise gate on each FFT channel. The channel is zeroed when it falls below a threshold set by taking the FFT of background noise. There's a lot more detail related to smoothing in time and frequency domains, and I think you should read the code to get the full story.

**Reply by JAJA-WHB** [reply]
Which part of source is Denoise?
